credentialdigger/client.py

```diff
@@ -1006,3 +1008,51 @@ def _get_scan_rules(self, category=None, exclude=None):
             raise ValueError('No rules found')
 
         return rules
+
+    def update_similar_snippets(self,
+                                target_snippet,
+                                state,
+                                repo_url,
+                                file_name=None,
+                                threshold=0.96):
+        """ Find snippets that are similar to the target
+        snippet and update their state.
+
+        Parameters
+        ----------
+        target_snippet: str
+        state: str
+            state to update similar snippets to
+        repo_url: str
+        file_name: str
+            restrict to a given file the search for similar snippets
+        threshold: double
+            update snippets with similarity score above threshold.
+            Values lesser than 0.94 do not generally imply any relevant
+            amount of similarity between snippets, and should
+            therefore not be used.
+
+        Returns
+        -------
+        int
+            The number of similar snippets found and updated
+        """
+
+        discoveries = self.get_discoveries(repo_url, file_name)
+        model = build_embedding_model()
+        """ Compute target snippet embedding """
+        target_snippet_embedding = compute_snippet_embedding(target_snippet,
+                                                             model)
+        n_updated_snippets = 0
+        for d in discoveries:
+            if d['state'] == 'new':
+                """ Compute snippet embedding """
+                snippet_embedding = compute_snippet_embedding(d['snippet'],
+                                                              model)
+                """ Compute similarity of target snippet and snippet """
+                similarity = compute_similarity(target_snippet_embedding,
+                                                snippet_embedding)
+                if similarity > threshold:
+                    n_updated_snippets += 1
+                    self.update_discovery(d['id'], state)
```

Comment:
does it work without passing the query like in the update_discovery method?

Reply:
Yes, because the update_discovery method that gets called is that of the child class, which doesn't require a query parameter. It definitely works.